galaxy_ng/tests/unit/api/test_api_ui_feature_flags.py

```diff
@@ -0,0 +1,23 @@
+from django.conf import settings
+
+from .base import BaseTestCase, get_current_ui_url
+
+
+class TestUiFeatureFlagsView(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.original_setting = settings.GALAXY_FEATURE_FLAGS
+        settings.GALAXY_FEATURE_FLAGS = {
+            'execution_environments': True,
+            'widget_x': False,
+        }
+
+    def tearDown(self):
+        super().tearDown()
+        settings.GALAXY_FEATURE_FLAGS = self.original_setting
+
+    def test_feature_flags_url(self):
+        self.feature_flags_url = get_current_ui_url('feature-flags')
+        response = self.client.get(self.feature_flags_url)
+        self.assertEqual(response.data['execution_environments'], True)
+        self.assertEqual(response.data['widget_x'], False)
```

Comment:
If feature flag default in code says it is on or off, I can test that one case. In the test I can change the setting for the second case, but I could not complete the test because I couldn't figure how to reload the url routes.

Reply:
hmm, interesting. I'm not sure how to reload urls either. Presumably if we did we could hook it to the settings changes signal.

I've never really used feature flags before so I don't know the expected behavior. If it's normal/expected that the feature flags are only 'applied' at application startup, then current code seems fine.

If expectation is to be able to flip them at run time, could be a problem.

Since feature-flags viewset is RO,  and are defined in settings.py, I'm assuming they are not intended to be changed dynamically at runtime. So probably fine as it is. If we run into problems in the test suites we can deal with it then.